client/packages/inventory/src/Stocktake/DetailView/ChangeLocationModal.tsx

```diff
@@ -69,24 +69,26 @@ export const ChangeLocationConfirmationModal = ({
       }
     >
       <Box gap={1} display="flex" flexDirection="column">
-        {hasMultipleItemsSelected && hasMultipleLocationTypes && (
+        {restrictedLocationTypeIds.length > 0 && (
           <Alert severity="warning" sx={{ width: 320 }}>
-            {hasMultipleLocationTypes
-              ? t('messages.cannot-change-location-multiple-types')
-              : t('messages.locations-restricted')}
+            {
+              hasConflictingRestrictedTypes
+                ? t('messages.cannot-change-location-multiple-types')
+                : t('messages.locations-restricted') // some lines `null`, some lines `freezer`, so we only show freezer locations - warn user
+            }
           </Alert>
         )}
         <InputWithLabelRow
           label={t('label.location')}
           labelWidth="100px"
           Input={
             <LocationSearchInput
-              disabled={false}
+              disabled={hasConflictingRestrictedTypes}
               selectedLocation={location}
               onChange={setLocation}
               width={210}
-              restrictedToLocationTypeId={restrictedToLocationTypeId}
-              enableAPI={!hasMultipleLocationTypes}
```

Comment:
This was cause of bug - if you load the location list, then go add some items so there is multiple/conflicting location types, it disabled the API. That meant we used the cached value, which was the full list.

Changed to just disable the input in this case